### DISCUSSION TITLE
v4 Access to options types

### DISCUSSION THREAD

**Discussion by Morb0** [telegraf/telegraf#1286] [Ideas] [answered]
How can I access `LaunchOptions` and `Options` under `Telegraf` namespace?
https://github.com/telegraf/telegraf/blob/develop/src/telegraf.ts#L40

In worst situation, of course, I can try to get these types from the constructor and parameters, but I would like to have an easier way.
```typescript
export type LaunchOptions = Parameters<Telegraf['launch']>[0];
export type Options = ConstructorParameters<typeof Telegraf>[1];
```

**Comment by wojpawlik** [comment]
Since de-exposing is breaking, I avoided exposing stuff for major release. I can easily expose them for 4.0.1 as `Telegraf.Options` and `Telegraf.LaunchOptions` if you need them, it's a 2 line change. My only concern is that `typedoc` would display the class and namespace separately:

![obraz](https://user-images.githubusercontent.com/23058303/104201151-b90d9700-5429-11eb-86cc-e1decd3e97d3.png)

If I exposed them at the top level as `TelegrafOptions` and `TelegrafLaunchOptions`, the names would just beg for namespacing.

**Comment by wojpawlik** [comment, marked as answer]
Fixed: `npm install telegraf/telegraf#57f16c70901e3c73b9ab67933a926fedf7a2b11d`
